core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -221,7 +221,8 @@ public BytesToBytesMap(
       SparkEnv.get() != null ? SparkEnv.get().blockManager() :  null,
       SparkEnv.get() != null ? SparkEnv.get().serializerManager() :  null,
       initialCapacity,
-      0.70,
+      // In order to re-use the longArray for sorting, the load factor cannot be larger than 0.5.
+      0.5,
```

Comment:
As analized in the comment, this only have about 10% (or less) difference in practice, I'd like not to have this complexity.